src/shims/unix/solarish/foreign_items.rs

```diff
@@ -69,6 +69,18 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                 this.write_null(dest)?;
             }
 
+            "pset_info" => {
+                let [_pset, _type, cpus, _list] =
+                    this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+                // We do not need to handle the current process cpu mask, available_parallelism
+                // implementation pass null anyway. We only care for the number of
+                // cpus.
+                // https://docs.oracle.com/cd/E88353_01/html/E37841/pset-info-2.html
+                let cpus = this.deref_pointer(cpus)?;
+                this.write_scalar(Scalar::from_u32(this.machine.num_cpus), &cpus)?;
+                this.write_null(dest)?;
+            }
```

Comment:
This shim could be called by other code outside of std, so it should do something reasonable also for the other parameters -- or it should be only used when `frame_in_std` is true.

Reply:
In general I would prefer if we didn't do the `frame_in_std` trick here. If someone gets an unsupported error in their code but their implementation is a copy of the standard library code, perhaps because it is a polyfill... That seems unnecessarily confusing. Can we `throw_unsup_format` except for the calls we need for the standard library? It sounds to me like we properly implement some subset of calls, unlike the `frame_in_std` we use for thread startup code.

Reply:
Seems like we'd have to check that type and list are null. And pset must be PS_MYID.

Reply:
Though FWIW `frame_in_std` is not just used for pre-init code. It also guards things like `getpwuid_r` as well; as a whole bunch of stuff on Windows. I agree it's potentially confusing, but IMO it's still better than no support at all.

Reply:
I agree that it's better than no support at all. I'll look at the other calls later to see if we're consistent with what I asked for above.